libminifi/include/Connection.h

```diff
@@ -28,174 +28,168 @@
+  // ! Get Connection Destination Processor
+  std::shared_ptr<core::Connectable> getDestination() {
+    return _destProcessor;
+  }
+  //! Set Connection relationship
+  void setRelationship(
+      core::Relationship relationship) {
+    relationship_ = relationship;
+  }
+  // ! Get Connection relationship
+  core::Relationship getRelationship() {
+    return relationship_;
+  }
+  //! Set Max Queue Size
+  void setMaxQueueSize(uint64_t size) {
+    _maxQueueSize = size;
+  }
+  //! Get Max Queue Size
+  uint64_t getMaxQueueSize() {
+    return _maxQueueSize;
+  }
+  //! Set Max Queue Data Size
+  void setMaxQueueDataSize(uint64_t size) {
+    _maxQueueDataSize = size;
+  }
+  //! Get Max Queue Data Size
+  uint64_t getMaxQueueDataSize() {
+    return _maxQueueDataSize;
+  }
+  //! Set Flow expiration duration in millisecond
+  void setFlowExpirationDuration(uint64_t duration) {
+    _expiredDuration = duration;
+  }
+  //! Get Flow expiration duration in millisecond
+  uint64_t getFlowExpirationDuration() {
+    return _expiredDuration;
+  }
+  //! Check whether the queue is empty
+  bool isEmpty();
+  //! Check whether the queue is full to apply back pressure
+  bool isFull();
+  //! Get queue size
+  uint64_t getQueueSize() {
+    std::lock_guard<std::mutex> lock(_mtx);
+    return queue_.size();
+  }
+  //! Get queue data size
+  uint64_t getQueueDataSize() {
+    return _maxQueueDataSize;
+  }
+  //! Put the flow file into queue
+  void put(std::shared_ptr<core::Record> flow);
+  //! Poll the flow file from queue, the expired flow file record also being returned
+  std::shared_ptr<core::Record> poll(
+      std::set<std::shared_ptr<core::Record>> &expiredFlowRecords);
+  //! Drain the flow records
+  void drain();
 
-private:
-	//! Mutex for protection
-	std::mutex _mtx;
-	//! Queued data size
-	std::atomic<uint64_t> _queuedDataSize;
-	//! Queue for the Flow File
-	std::queue<FlowFileRecord *> _queue;
-	//! Logger
-	std::shared_ptr<Logger> logger_;
-	// Prevent default copy constructor and assignment operation
-	// Only support pass by reference or pointer
-	Connection(const Connection &parent);
-	Connection &operator=(const Connection &parent);
+  void yield() {
 
-};
+  }
+
+  bool isWorkAvailable() {
+    return !isEmpty();
+  }
+
+  bool isRunning() {
+    return true;
+  }
 
+ protected:
+  //! Source Processor UUID
+  uuid_t _srcUUID;
+  //! Destination Processor UUID
+  uuid_t _destUUID;
+  //! Relationship for this connection
+  core::Relationship relationship_;
+  //! Source Processor (ProcessNode/Port)
+  std::shared_ptr<core::Connectable> _srcProcessor;
+  //! Destination Processor (ProcessNode/Port)
+  std::shared_ptr<core::Connectable> _destProcessor;
+  //! Max queue size to apply back pressure
+  std::atomic<uint64_t> _maxQueueSize;
+  //! Max queue data size to apply back pressure
+  std::atomic<uint64_t> _maxQueueDataSize;
+  //! Flow File Expiration Duration in= MilliSeconds
+  std::atomic<uint64_t> _expiredDuration;
```

Comment:
Can you move "_" to end of class data member to adhere to Google C++ Style guid? Same for all variables here.